### PR TITLE
Dial Press and Rotate Streamdeck+

### PR DESCRIPTION
**Pull request by K0jiki**
Hi, I added dialPress and Rotate event Handling to your library.
As for now I'm still working on setFeedback implementation, which seems to work a different way than showOk or sendToPlugin.

Because of that I did not include a new example for Dial Actions, I will include them when I get TouchScreen feedback working.

Thank for taking your time reviewing my PR !

### CODE REVIEW

File: StreamDeckSDK/ESDSDKDefines.h [hunk 1]
```diff
@@ -159,5 +163,6 @@ enum {
   kESDSDKDeviceType_StreamDeckXL = 2,
   kESDSDKDeviceType_StreamDeckMobile = 3,
   kESDSDKDeviceType_CorsairGKeys = 4,
-  kESDSDKDeviceType_StreamDeckPedal = 5
+  kESDSDKDeviceType_StreamDeckPedal = 5,
+  kESDSDKDeviceType_StreamDeckPlus = 7 // From volume controller manifest.json
```
Comment (fredemmott) [inline StreamDeckSDK/ESDSDKDefines.h:167]: For my reference, 6 should be kESDSDKDeviceType_CorsairVoyager

File: StreamDeckSDK/ESDAction.cpp [hunk 2]
```diff
@@ -39,6 +39,30 @@ void ESDAction::KeyDown(const nlohmann::json& settings) {
 void ESDAction::KeyUp(const nlohmann::json& settings) {
 }
 
+void ESDAction::DialUp(const nlohmann::json& settings) {
+
+}
+
+void ESDAction::DialDown(const nlohmann::json& settings) {
+
+}
```
Comment (fredemmott) [inline StreamDeckSDK/ESDAction.cpp:48]: The naming here's consistent, but a bit confusing for a dial.

What's the physical interaction? Is it a button push, or e.g. a capacitive touch sensor?

If it's a touch sensor, perhaps `TouchStart` and `TouchEnd` instead?

**Review by fredemmott — COMMENTED** [review]
Thanks - this is great, just a a naming nit

Also, could you run clang-format? If you're in vscode, 'format document' will do this

File: StreamDeckSDK/ESDAction.cpp (hunk 2 above)
Comment (K0jiki) [reply StreamDeckSDK/ESDAction.cpp:48]: It's a button press on the Dial, you can rotate and press them.
There's also a touch screen but I did not start working on it.

File: StreamDeckSDK/ESDSDKDefines.h (hunk 1 above)
Comment (K0jiki) [reply StreamDeckSDK/ESDSDKDefines.h:167]: I see, I will add that.

**Comment by fredemmott** [comment]
Thanks!

**Comment by fredemmott** [comment]
@jorysbdv I'm starting to try to use this, and thanks to Elgato's decision, there's not really an API approach that I'm completely happy with - i've put my thoughts on https://github.com/fredemmott/StreamDeck-CPPSDK/issues/21 and would welcome your feedback

**Comment by jorysbdv** [comment]
Hi thanks for asking me. I gave you an opinion your issue #21. Have a great
evening!

Le sam. 25 mars 2023, 17:17, Fred Emmott ***@***.***> a
écrit :

**Pull request merged**